protocol/message_persistence.go

```diff
@@ -1891,6 +1891,73 @@ func (db sqlitePersistence) MarkMessagesSeen(chatID string, ids []string) (uint6
 	return countWithMentions + countNoMentions, countWithMentions, err
 }
 
+func (db sqlitePersistence) MarkMessageAsUnread(chatID string, messageID string) (uint64, uint64, error) {
```

Comment:
just suggestion: you can parametrise `MarkMessagesSeen` with boolean seen/unseen flag instead of this function

Reply:
MarkAsUnread is more than just making a bunch of items as unread. The difference is that with MarkAsUnread, we also need to mark every items with smaller timestamp than the current element as read. The name mark as read is still valid though. So that reason I think a separate function makes more sense.